knack/config.py

```diff
@@ -109,9 +109,9 @@ def sections(self):
 
     def items(self, section):
         import re
-        pattern = self.env_var_name(section, '.+')
-        candidates = [(k.split('_')[-1], os.environ[k], k) for k in os.environ if re.match(pattern, k)]
-        result = {c[0]: c for c in candidates}
+        pattern = self.env_var_name(section, '[0-9A-Z][0-9A-Z_]*')
```

Comment:
This will match `CLI_MYSECTION_T` part in `CLI_MYSECTION_Test_Option`, defeating our purpose of ignoring invalid env options like `CLI_MYSECTION_Test_Option`.